backend/src/services/groups-to-permissions/groups-to-permissions.service.ts

```diff
@@ -26,6 +26,55 @@ class GroupsToPermissions {
       permissionId: model.permissionId,
     };
   }
+
+  async getPermissionsByGroupId(
+    groupId: number,
+  ): Promise<GroupsToPermissionsResponseDto[]> {
+    const permissions = await this.#groupsToPermissionsRepository.getByGroupId(
+      groupId,
+    );
+
+    return permissions;
+  }
+
+  async updatePermissionsByGroupId(groupsToPermissions: {
+    groupId: number;
+    permissionIds: number[];
+  }): Promise<void> {
+    const { groupId, permissionIds } = groupsToPermissions;
+    const groupPrermissions = await this.getPermissionsByGroupId(groupId);
+    const dbPermissionIds = groupPrermissions.map(
+      (permission) => permission.permissionId,
+    );
+
+    await Promise.all(
+      permissionIds.map((permissionId: number) => {
+        if (!dbPermissionIds.includes(permissionId)) {
+          return this.createGroupsToPermissions({
+            groupId,
+            permissionId,
+          });
+        }
+      }),
```

Comment:
You can split up permissionIds to update separately from calling request to repository method to get `permissionsToCreate` and `permissionsToDelete` - this will make it more simple to understand what's going on in db requests
```suggestion
      permissionsToCreate.map((permissionId: number) => {
        return this.createGroupsToPermissions({
            groupId,
            permissionId,
          });
      }),
```

Reply:
agreed with all Andrey's comments